packages/cli/templates/defaults/app/route-templates/xml.tsx

```diff
@@ -0,0 +1,61 @@
+/* eslint-disable camelcase */
+import { type LoaderFunctionArgs, redirect } from "@remix-run/server-runtime";
+import { ReactSdkContext } from "@webstudio-is/react-sdk";
+import { Page } from "../../../../__generated__/_index";
+import {
+  loadResources,
+  getPageMeta,
+  getRemixParams,
+} from "../../../../__generated__/_index.server";
+
+import { assetBaseUrl, imageBaseUrl, imageLoader } from "../constants.mjs";
+import { renderToString } from "react-dom/server";
+
+export const loader = async (arg: LoaderFunctionArgs) => {
+  const url = new URL(arg.request.url);
+  const host =
+    arg.request.headers.get("x-forwarded-host") ||
+    arg.request.headers.get("host") ||
+    "";
+  url.host = host;
+  url.protocol = "https";
+
+  const params = getRemixParams(arg.params);
+
+  const system = {
+    params,
+    search: Object.fromEntries(url.searchParams),
+    origin: url.origin,
+  };
+
```

Comment:
so every new system variable we will add will have to be added in all of these places and probably more? I think we already now know there will be many more system variables and we need to use a function

Additionally would be nice if system had a ts type System

Reply:
1. turns out type System already exists in the builder
2. Reason why I think you can't just add another system like this without adding a type is because now we can't be sure where are all the places where we need to also add origin, it can fail without a warning

Reply:
all good here, loadResources and getPageMeta validate it

Reply:
according to @TrySound its being checked in the function we pass it to